src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/MessageProcessor.java

```diff
@@ -0,0 +1,59 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services;
+
+import com.microsoft.azure.servicebus.IMessage;
+import com.microsoft.azure.servicebus.IMessageHandler;
+import com.microsoft.azure.servicebus.IMessageReceiver;
+import com.microsoft.azure.servicebus.primitives.ServiceBusException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.ReceiverProvider;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.exceptions.ConnectionException;
+
+@Service
+public class MessageProcessor {
+    public static final String TEST_MSG_LABEL = "test";
+
+    private static final Logger logger = LoggerFactory.getLogger(MessageProcessor.class);
+
+    private final ReceiverProvider receiverProvider;
+    private final IMessageHandler envelopeProcessor;
+
+    public MessageProcessor(ReceiverProvider receiverProvider, IMessageHandler envelopeProcessor) {
+        this.receiverProvider = receiverProvider;
+        this.envelopeProcessor = envelopeProcessor;
+    }
+
+    @Scheduled(fixedDelayString = "${queue.read-interval}")
+    public void run() {
+        IMessage msg = null;
+        try {
+            IMessageReceiver msgReceiver = receiverProvider.get();
+            while ((msg = msgReceiver.receive()) != null) {
+                process(msg);
+                msgReceiver.complete(msg.getLockToken());
+            }
+        } catch (ServiceBusException | ConnectionException exc) {
+            logger.error("Unable to read messages from queue", exc);
+        } catch (InterruptedException exc) {
+            Thread.currentThread().interrupt();
+            logger.error("Unable to read messages from queue", exc);
+        } catch (Exception e) {
+            logger.error("Error processing message ID: {}", (msg != null) ? msg.getMessageId() : "none", e);
+        }
+    }
+
+    private void process(IMessage msg) throws InterruptedException, java.util.concurrent.ExecutionException {
+        if (isTestMessage(msg)) {
+            logger.info("Received test message, messageId: {}", msg.getMessageId());
+        } else {
+            envelopeProcessor.onMessageAsync(msg).get();
```

Comment:
yup ... as explained above and after a chat with rafal yesterday I think it might be best if we use the api for the receiving of the messages.  
Unless we have a reason not to ?

Reply:
yes I agree we need to leverage the framework. Can we register the message handler as shown in the example rather than calling the method ?

Reply:
The whole point of registering (which is not yet registered!) handler callback is to get rid of this schedule and all this try/catch nonsense. If we still going with scheduled approach there was no need to introduce concurrent callback approach. I would strongly recommend to use callback + register. this super simplifies everything